test/unit/store/state_db_test.exs

```diff
@@ -0,0 +1,135 @@
+defmodule Unit.Store.StateDb do
+  alias Fixtures.Random
+  alias LambdaEthereumConsensus.Store.StateDb
+  alias Types.BeaconState
+  alias Types.StateInfo
+
+  use ExUnit.Case
+
+  setup_all do
+    Application.fetch_env!(:lambda_ethereum_consensus, ChainSpec)
+    |> Keyword.put(:config, MinimalConfig)
+    |> then(&Application.put_env(:lambda_ethereum_consensus, ChainSpec, &1))
+  end
+
+  setup %{tmp_dir: tmp_dir} do
+    start_link_supervised!({LambdaEthereumConsensus.Store.Db, dir: tmp_dir})
+    :ok
+  end
+
+  defp get_state_info() do
+    {:ok, encoded} =
+      File.read!("test/fixtures/validator/proposer/beacon_state.ssz_snappy")
+      |> :snappyer.decompress()
+
+    {:ok, decoded} = SszEx.decode(encoded, BeaconState)
+    {:ok, state_info} = StateInfo.from_beacon_state(decoded)
+    state_info
+  end
+
+  # Returns a new `state_info` with:
+  # - `slot` incremented by 1.
+  # - A random `block_root`.
+  # - A random `state_root`.
+  defp modify_state_info(state_info) do
+    new_beacon_state = state_info.beacon_state |> Map.put(:slot, state_info.beacon_state.slot + 1)
+    {:ok, new_state_info} = StateInfo.from_beacon_state(new_beacon_state)
+
+    new_state_info
+    |> Map.put(:block_root, Random.root())
+    |> Map.put(:root, Random.root())
```

Comment:
Is the second `Map.put` necessary? Isn't the root recalculated when calling `StateInfo.from_beacon_state/1`?

Reply:
You are right! Thank you!